File: 0953-verifying-an-alien-dictionary/0953-verifying-an-alien-dictionary.py
class Solution:
    def isAlienSorted(self, words: List[str], order: str) -> bool:
        orderMap = [0]*26
        
        for i in range(len(order)):
            orderMap[ord(order[i])-ord('a')] = i;
        
        for i in range(len(words)-1):
            for j in range(len(words[i])):
                if(j >= len(words[i+1])):
                    return False
                
                if(words[i][j] != words[i+1][j]):
                    curr = ord(words[i][j])-ord('a');
                    next = ord(words[i+1][j])-ord('a');
                    
                    if orderMap[curr] > orderMap[next]:
                        return False
                    else:
                        break
                
        return True    

File: 0953-verifying-an-alien-dictionary/0953-verifying-an-alien-dictionary.cpp
class Solution {
public:
    bool isAlienSorted(vector<string>& words, string order) {
        vector<int> orderMap(26);
        for(int i =0;i < order.size();i++){
            orderMap[order[i]-'a'] = i;
        }
        
        for(int i = 0;i < words.size()-1;i++){
            for(int j = 0;j < words[i].size();j++){
                if(j >= words[i+1].size())
                    return false;
                
                if(words[i][j] != words[i+1][j]){
                    int curr = words[i][j]-'a';
                    int next = words[i+1][j]-'a';
                    
                    if(orderMap[curr]>orderMap[next])
                        return false;
                    else
                        break;
                }
            }
        }
        return true;
    }
};